### PR TITLE
replace snowflaked to avoid conflicts

### PR DESCRIPTION
**Pull request by gallone2000**
### Overview
This PR removes the external `snowflaked` dependency and replaces it with a custom, lock-free snowflake ID generator implementation. The new implementation maintains **full behavioral compatibility** with `snowflaked 1.0.3` while providing equivalent or improved performance characteristics.

### Key Changes
- **Removed dependency**: Moved `snowflaked` from production to dev-dependencies (used only for testing/oracle validation)
- **Custom generator**: Implemented a deterministic, thread-safe snowflake generator using atomic CAS operations with Relaxed ordering
- **Bit layout**: Maintains the exact same 42/10/12 bit partitioning (timestamp/instance/sequence) as the original crate
- **Behavior parity**: Clock rollback handling, sequence overflow with millisecond wait, and timestamp extraction match the oracle exactly

### Test Coverage
I implemented comprehensive test suites to ensure robustness and guard against future regressions:

#### Deterministic Tests
- Single and multi-threaded uniqueness and monotonicity
- System time extraction and Unix epoch alignment
- Millisecond boundary transitions
- u63 boundary handling
- JSON serialization roundtrips
- Model-based scripted state transitions with controlled clock

#### Oracle-Based Tests
- Bit-for-bit packing equivalence with `snowflaked::Snowflake::from_parts`
- Failure mode validation: both implementations panic identically on clock rollback

#### Property-Based Tests
- Generated with `proptest` to explore large input spaces automatically
- Roundtrip pack/unpack preservation of components
- Boundary-biased strategies covering edge cases (0, max, near-max values)
- Monotonicity and uniqueness under varying sample sizes
- u63 boundary behavior for all valid input combinations

Includes CI-friendly seed control via `PROPTEST_SEED` env var for deterministic property test execution.

### Performance

Benchmark suite included for regression detection.

### Rationale
The custom implementation:
1. Eliminates an external dependency while maintaining full compatibility
2. Provides transparent behavior through extensive oracle testing
3. Reduces operational risk via property-based testing of edge cases
4. Scales equally or better under concurrent load

### CODE REVIEW

File: opsqueue/src/common/submission.rs [hunk 1]
```diff
@@ -137,6 +236,526 @@ impl From<&SubmissionId> for std::time::SystemTime {
     }
 }
 
+#[cfg(test)]
+mod snowflake_tests {
```
Comment (Qqwy) [inline opsqueue/src/common/submission.rs]: Nit: I think it's great that we test a big part of the API/conformance of the snowflake IDs. However, since the System Under Test is only the snowflake IDs and that is only loosely connected with submissions as a whole, I recommend moving both the functions implementing `generate_submission_snowflake` and the related tests to a separate dedicated file.

**Review by Qqwy — APPROVED** [review]

File: opsqueue/benches/snowflake_compare.rs [hunk 2]
```diff
@@ -0,0 +1,95 @@
+use std::hint::black_box;
+use std::time::Instant;
+
+use criterion::{criterion_group, criterion_main, Criterion};
+use opsqueue::common::submission::SubmissionId;
+use snowflaked::sync::Generator;
+
+static ORACLE_GENERATOR: Generator = Generator::new(0);
+
+fn bench_single_thread(c: &mut Criterion) {
```
Comment (jerbaroo) [inline opsqueue/benches/snowflake_compare.rs:11]: Suggest add (very short is fine) docstrings to these functions explaining what they do ?

File: opsqueue/src/common/snowflake.rs [hunk 3]
```diff
@@ -0,0 +1,662 @@
...
+    }
+}
+
+#[cfg(test)]
+fn generate_submission_snowflake_with<Now, Tick>(
+    state: &AtomicU64,
+    now_ms: Now,
+    tick_wait: Tick,
+) -> u64
+where
```
Comment (jerbaroo) [inline opsqueue/src/common/snowflake.rs:94]: There seems to be a good bit of duplicated logic across `generate_submission_snowflake_with`, `generate_submission_snowflake`, `next_with_tick`. Maybe we can generalize and implement the others in terms of one?

File: opsqueue/src/common/snowflake.rs [hunk 4]
```diff
@@ -0,0 +1,662 @@
...
+    }
+}
+
+pub(crate) fn generate_submission_snowflake() -> u64 {
+    let instance = SUBMISSION_SNOWFLAKE_INSTANCE;
+
+    loop {
+        let last = LAST_SUBMISSION_ID.load(Ordering::Relaxed);
+        let last_timestamp_ms = snowflake_timestamp_ms(last);
+        let last_sequence = last & SEQUENCE_MASK;
```
Comment (jerbaroo) [inline opsqueue/src/common/snowflake.rs]: This seems to be the same as `snowflake_sequence`, suggest call that ?

File: opsqueue/src/common/snowflake.rs [hunk 5]
```diff
@@ -0,0 +1,662 @@
...
+        let now_ms_value = now_unix_ms();
+
+        let next = if now_ms_value < last_timestamp_ms {
+            panic!("Clock has moved backwards! This is not supported");
+        } else if now_ms_value > last_timestamp_ms {
+            (now_ms_value << TIMESTAMP_SHIFT) | (instance << SEQUENCE_BITS)
```
Comment (jerbaroo) [inline opsqueue/src/common/snowflake.rs]: Should we use `pack_submission_snowflake` ?

**Review by jerbaroo — APPROVED** [review]
LGTM! A few small comments for your consideration.

File: opsqueue/src/common/snowflake.rs (hunk 3 above)
Comment (gallone2000) [reply opsqueue/src/common/snowflake.rs:94]: Good call, thanks. I refactored this to reduce duplication between `generate_submission_snowflake_with` and `generate_submission_snowflake`: both now delegate to a shared `next_submission_snowflake(...)` core, while keeping separate wrappers for test/production time-wait wiring.

I intentionally kept `next_with_tick` in the test model separate, so the model remains an independent oracle instead of reusing production logic.

File: opsqueue/src/common/snowflake.rs (hunk 4 above)
Comment (gallone2000) [reply opsqueue/src/common/snowflake.rs]: Good catch — agreed. This was addressed by the refactor: the duplicated `last & SEQUENCE_MASK` path is gone, and sequence extraction now goes through `snowflake_sequence(last)` in the shared core logic.

File: opsqueue/src/common/snowflake.rs (hunk 5 above)
Comment (gallone2000) [reply opsqueue/src/common/snowflake.rs]: Agreed — good suggestion. I switched this to use `pack_submission_snowflake(...)` so packing is centralized and consistent (no inline bit-packing logic left outside that helper).

**Review by jerbaroo — APPROVED** [review]

**Comment by gallone2000** [comment]
@OpsBotPrime merge and deploy to production

**Comment by OpsBotPrime** [comment]
Unknown or invalid command found:

    comment:1:30:
      |
    1 | @OpsBotPrime merge and deploy to production
      |                              ^
    No deployment environments have been configured.
[Basic usage is explained here.](https://github.com/channable/hoff/blob/master/readme.md#using-hoff)

**Comment by gallone2000** [comment]
@OpsBotPrime merge and deploy

**Comment by OpsBotPrime** [comment]
Unknown or invalid command found:

    comment:1:30:
      |
    1 | @OpsBotPrime merge and deploy
      |                              ^
    No deployment environments have been configured.
[Basic usage is explained here.](https://github.com/channable/hoff/blob/master/readme.md#using-hoff)

**Comment by gallone2000** [comment]
@OpsBotPrime merge

**Comment by OpsBotPrime** [comment]
Rebased as eb3442c18dc9e68312e697748db6f6ca0a27ff8f, waiting for CI …

**Comment by OpsBotPrime** [comment]
[CI job :yellow_circle:](https://channable.semaphoreci.com/workflows/a71f6dfd-b1d7-4239-b5a7-9c3fad0126eb?pipeline_id=0ff0788e-7eb7-40e0-bfc2-524b9a2d6374) started.

**Comment by OpsBotPrime** [comment]
The [build failed :x:](https://channable.semaphoreci.com/workflows/a71f6dfd-b1d7-4239-b5a7-9c3fad0126eb?pipeline_id=0ff0788e-7eb7-40e0-bfc2-524b9a2d6374).

If this is the result of a flaky test, then tag me again with the `retry` command.  Otherwise, push a new commit and tag me again.

**Comment by gallone2000** [comment]
@OpsBotPrime merge and tag

**Comment by OpsBotPrime** [comment]
Rebased as a39ab528a03933b874c2a0dc7ef14c6296d036f8, waiting for CI …

**Comment by OpsBotPrime** [comment]
[CI job :yellow_circle:](https://channable.semaphoreci.com/workflows/e62a0dcb-9e33-448e-a782-58e0cc90cb41?pipeline_id=dc515663-df9f-4b65-bc33-22030808e105) started.

**Comment by OpsBotPrime** [comment]
The [build failed :x:](https://channable.semaphoreci.com/workflows/e62a0dcb-9e33-448e-a782-58e0cc90cb41?pipeline_id=dc515663-df9f-4b65-bc33-22030808e105).

If this is the result of a flaky test, then tag me again with the `retry` command.  Otherwise, push a new commit and tag me again.

**Comment by gallone2000** [comment]
@OpsBotPrime  retry

**Comment by OpsBotPrime** [comment]
Rebased as 4fe6fc26d62be69770f0d4936513f22dd878b6d1, waiting for CI …

**Comment by OpsBotPrime** [comment]
[CI job :yellow_circle:](https://channable.semaphoreci.com/workflows/b780aae9-d55a-405a-9417-278d0db0e30d?pipeline_id=684b18ce-3a0e-4a7f-80d5-ac95edd336de) started.

**Comment by OpsBotPrime** [comment]
@gallone2000 I tagged your PR with [v55](https://github.com/channable/opsqueue/releases/tag/v55). Please wait for the build of 4fe6fc26d62be69770f0d4936513f22dd878b6d1 to pass and don't forget to deploy it!
